### PR TITLE
Bm60poker updates prepping for Via support

### PR DESCRIPTION
**Pull request by ipetepete**
## Description

- Changed the `device_id` in config to be unique, previously using the BM60RGB device_id.
- Added Via keymap
- Updated info.json to be usable for Via
- Updated my personal keymap

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bm60poker/config.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B50 // "KP"
+#define PRODUCT_ID      0xEF8D
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    KP Republic
+#define PRODUCT         KP Republic BM60 RGB POKER
+#define DESCRIPTION     An RGB 60% hotswappable standard ansi layout board
```
Comment (zvecr) [inline keyboards/bm60poker/config.h:28]:
```suggestion
```

File: keyboards/bm60poker/config.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2020 REPLACE_WITH_YOUR_NAME
```
Comment (zvecr) [inline keyboards/bm60poker/config.h:2]:
```suggestion
Copyright 2020 ipetepete
```

File: keyboards/bm60poker/info.json [hunk 3]
```diff
@@ -0,0 +1,153 @@
+{
```
Comment (zvecr) [inline keyboards/bm60poker/info.json:1]: This file is the wrong format, via config goes to <https://github.com/the-via/keyboards>

File: keyboards/bm60poker/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/* Copyright 2020 REPLACE_WITH_YOUR_NAME
```
Comment (zvecr) [inline keyboards/bm60poker/keymaps/via/keymap.c:1]:
```suggestion
/* Copyright 2020 ipetepete
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by zvecr** [comment]
Conflicts will also need to be resolved before this can be merged.

File: keyboards/bm60poker/config.h [hunk 5]
```diff
@@ -21,10 +21,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 /* USB Device descriptor parameter */
 #define VENDOR_ID       0x4B50 // "KP"
-#define PRODUCT_ID      0xEF8E
+#define PRODUCT_ID      0xEF8D
 #define DEVICE_VER      0x0001
 #define MANUFACTURER    KP Republic
-#define PRODUCT         BM60POKER
+#define PRODUCT         KP Republic BM60 RGB POKER
```
Comment (fauxpark) [inline keyboards/bm60poker/config.h:27]:
```suggestion
#define PRODUCT         BM60 RGB POKER
```

File: keyboards/bm60poker/config.h [hunk 6]
```diff
@@ -68,3 +68,6 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 //     #define RGBLIGHT_EFFECT_RGB_TEST
 //     #define RGBLIGHT_EFFECT_ALTERNATING
 #endif
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (fauxpark) [inline keyboards/bm60poker/config.h:73]:
```suggestion
```

File: keyboards/bm60poker/config.h [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /*
 Copyright 2020 ipetepete
 
-This program is free software: you can redistribute it and/or modify
+	This program is free software: you can redistribute it and/or modify
```
Comment (fauxpark) [inline keyboards/bm60poker/config.h:4]:
```suggestion
This program is free software: you can redistribute it and/or modify
```

File: keyboards/bm60poker/info.json [hunk 8]
```diff
@@ -1,74 +0,0 @@
-{
```
Comment (fauxpark) [inline keyboards/bm60poker/info.json:1]: Please restore this file.

File: keyboards/bm60poker/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        KC_LSFT,            KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,
+        KC_LCTL,   KC_LGUI,   KC_LALT,                       KC_SPC,                              KC_RALT,   KC_RGUI,   MO(1),     KC_RCTL
+    ),
+    [1] = LAYOUT(
+        RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+        _______,     RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______, _______,
+        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,            _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,   _______,   _______,                      _______,                              _______,   _______,   _______,   _______
+    ),
```
Comment (fauxpark) [inline keyboards/bm60poker/keymaps/via/keymap.c:32]: Needs two extra empty layers.

File: keyboards/bm60poker/rules.mk [hunk 10]
```diff
@@ -7,18 +7,23 @@ BOOTLOADER = atmel-dfu
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
-MOUSEKEY_ENABLE = yes       # Mouse keys
+BOOTMAGIC_ENABLE = no     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:11]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
```

File: keyboards/bm60poker/rules.mk [hunk 11]
```diff
@@ -7,18 +7,23 @@ BOOTLOADER = atmel-dfu
...
 EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:14]:
```suggestion
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
```

File: keyboards/bm60poker/rules.mk [hunk 12]
```diff
@@ -7,18 +7,23 @@ BOOTLOADER = atmel-dfu
...
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-BLUETOOTH_ENABLE = no       # Enable Bluetooth
-AUDIO_ENABLE = no           # Audio output
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:25]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/bm60poker/rules.mk [hunk 13]
```diff
@@ -7,18 +7,23 @@ BOOTLOADER = atmel-dfu
...
 RGB_MATRIX_ENABLE = WS2812
-LAYOUTS = 60_ansi
+
+# generated by KBFirmware JSON to QMK Parser
+# https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:29]:
```suggestion
LAYOUTS = 60_ansi
```

File: keyboards/planck/keymaps/ipetepete/config.h [hunk 14]
```diff
@@ -0,0 +1,3 @@
+#pragma once
+//#define BACKLIGHT_PIN B7
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ipetepete/config.h:2]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Your personal keymaps should really be submitted as a separate PR.

File: keyboards/bm60poker/info.json (hunk 8 above)
Comment (ipetepete) [reply keyboards/bm60poker/info.json:1]: I'm getting conflicting information about this file @zvecr stated it didn't belong in this repo: https://github.com/qmk/qmk_firmware/pull/10987#discussion_r528228791

File: keyboards/bm60poker/rules.mk [hunk 15]
```diff
@@ -5,20 +5,20 @@ MCU = atmega32u4
 BOOTLOADER = atmel-dfu
 
 # Build Options
-#   change yes to no to disable
+# change yes to no to disable
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:8]:
```suggestion
#   change yes to no to disable
```

File: keyboards/bm60poker/rules.mk [hunk 16]
```diff
@@ -5,20 +5,20 @@ MCU = atmega32u4
...
-BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
-MOUSEKEY_ENABLE = yes       # Mouse keys
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
+BOOTMAGIC_ENABLE = no     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes     # Mouse keys
+EXTRAKEY_ENABLE = yes     # Audio control and System control
+CONSOLE_ENABLE = no       # Console for debug
+COMMAND_ENABLE = no       # Commands for debug and configuration
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:14]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
```

File: keyboards/bm60poker/rules.mk [hunk 17]
```diff
@@ -5,20 +5,20 @@ MCU = atmega32u4
...
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
-SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+SLEEP_LED_ENABLE = no     # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:16]:
```suggestion
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```

File: keyboards/bm60poker/rules.mk [hunk 18]
```diff
@@ -5,20 +5,20 @@ MCU = atmega32u4
...
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-BLUETOOTH_ENABLE = no       # Enable Bluetooth
-AUDIO_ENABLE = no           # Audio output
+NKRO_ENABLE = no          # USB Nkey Rollover
+BACKLIGHT_ENABLE = no     # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no      # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no     # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no         # Audio output on port C6
 RGB_MATRIX_ENABLE = WS2812
 LAYOUTS = 60_ansi
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:24]:
```suggestion
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = WS2812

LAYOUTS = 60_ansi
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bm60poker/rules.mk [hunk 19]
```diff
@@ -5,20 +5,20 @@ MCU = atmega32u4
 BOOTLOADER = atmel-dfu
 
 # Build Options
-#   change yes to no to disable
+#  change yes to no to disable
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:8]:
```suggestion
#   change yes to no to disable
```

File: keyboards/bm60poker/rules.mk [hunk 20]
```diff
@@ -5,20 +5,20 @@ MCU = atmega32u4
...
-BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
-MOUSEKEY_ENABLE = yes       # Mouse keys
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes      # Mouse keys
+EXTRAKEY_ENABLE = yes      # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:14]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
```

File: keyboards/bm60poker/rules.mk [hunk 21]
```diff
@@ -5,20 +5,20 @@ MCU = atmega32u4
...
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
-SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+SLEEP_LED_ENABLE = no      # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:16]:
```suggestion
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```

File: keyboards/bm60poker/rules.mk [hunk 22]
```diff
@@ -5,20 +5,20 @@ MCU = atmega32u4
...
-NKRO_ENABLE = no            # USB Nkey Rollover
-BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-BLUETOOTH_ENABLE = no       # Enable Bluetooth
-AUDIO_ENABLE = no           # Audio output
+NKRO_ENABLE = no           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no      # Enable Bluetooth
+AUDIO_ENABLE = no          # Audio output
```
Comment (fauxpark) [inline keyboards/bm60poker/rules.mk:22]:
```suggestion
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/preonic/keymaps/ipetepete/config.h [hunk 23]
```diff
@@ -0,0 +1,13 @@
+#pragma once
```
Comment (drashna) [inline keyboards/preonic/keymaps/ipetepete/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by ipetepete** [comment]
I've removed my personal keymaps, should I just close this request and issue a new one? I think I've complied with all the change requests but this PR is little dated....idk, I'll go with whatever the consensus is.

**Comment by fauxpark** [comment]
@ipetepete you only need to remove your keymaps from this PR, and submit them as a separate one. The BM60 changes are still fine.

**Comment by ipetepete** [comment]
@faupark I reverted my personal keymaps to the ones in the master branch.

File: keyboards/preonic/keymaps/ipetepete/keymap.c [hunk 24]
```diff
@@ -0,0 +1,285 @@
...
+  uint16_t held_keycode_timer = timer_read();
+
+  switch(biton32(layer_state)){
+    case _LOWER:
+      if(clockwise){
+        register_code(KC_MNXT);
+        while(timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY){
+          // noop
+        }
+        unregister_code(KC_MNXT);
```
Comment (drashna) [inline keyboards/preonic/keymaps/ipetepete/keymap.c:183]:
```suggestion
        tap_code(KC_MNXT);
```

Tap code already has a wait, based on the `TAP_CODE_DELAY` define.

File: keyboards/preonic/keymaps/ipetepete/keymap.c [hunk 25]
```diff
@@ -0,0 +1,285 @@
...
+      }else{
+        register_code(KC_MPRV);
+        while(timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY){
+          // noop
+        }
+        unregister_code(KC_MPRV);
```
Comment (drashna) [inline keyboards/preonic/keymaps/ipetepete/keymap.c:189]:
```suggestion
        tap_code(KC_MPRV);
```

File: keyboards/preonic/keymaps/ipetepete/keymap.c [hunk 26]
```diff
@@ -0,0 +1,285 @@
...
+          } else {
+            muse_offset--;
+          }
+      }else{
+        if(clockwise){
+          register_code(KC_PGUP);
+          while(timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY){
+            // noop
+          }
+          unregister_code(KC_PGUP);
```
Comment (drashna) [inline keyboards/preonic/keymaps/ipetepete/keymap.c:206]:
```suggestion
        tap_code(KC_PGUP);
```

File: keyboards/preonic/keymaps/ipetepete/keymap.c [hunk 27]
```diff
@@ -0,0 +1,285 @@
...
+        }else{
+            register_code(KC_PGDN);
+            while(timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY){
+              // noop
+            }
+            unregister_code(KC_PGDN);
```
Comment (drashna) [inline keyboards/preonic/keymaps/ipetepete/keymap.c:212]:
```suggestion
        tap_code();
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by fauxpark** [comment]
@ipetepete your keymaps are still present in this PR.

**Comment by ipetepete** [comment]
I'm closing this pull request and submitting a new clean request.

**Pull request closed without merging**